pages/concepts/presence.mdx

```diff
@@ -20,45 +20,51 @@ r.subscribeToPresence((presentClientIDs) => {
   `presentClientIDs` will always contain the client's own client ID.
 </Callout>
 
-## Per-Client Presence State
+## Presence State
 
-The `subscribeToPresence` method only tells you the IDs of the clients that are present.
-To do something useful with them, you will combine these IDs with additional synced data about clients.
+The `subscribeToPresence` method only tells you the client IDs of the clients that are present.
+To do something useful with them, you will combine these client IDs with additional synced data about clients.
 This additional client data is custom to your application. It is read and written
 like any other Reflect data and can be [modeled using the Rails helper library](./model).
 
 For example you might model your client data like so:
 
 ```ts
-import { generate } from "@rocicorp/rails";
+import { generatePresence } from "./generate-presence.js";
 
 type Client = {
+  clientID: string;
   id: string;
   name: string;
   cursor: { x: number; y: number };
 };
 
 export const {
-  put: putClient,
+  set: setClient,
   get: getClient,
   init: initClient,
   update: updateClient,
   list: listClients,
-} = generate<Client>("client");
+  listEntries: listClientEntries,
+} = generatePresence<Client>("client");
 ```
 
 You can then filter this client data to just render the currently connected clients:
 
 ```ts
-let presentClientIDs: ClientID[] = [];
+let presentClientIDs: readonly ClientID[] = [];
 r.subscribeToPresence((clientIDs) => {
   presentClientIDs = clientIDs;
 });
 
 let clients: Record<ClientID, Client> = {};
 r.subscribe(
   async (tx) => {
-    return Object.fromEntries(await listClientEntries(tx));
+    const result: Record<ClientID, Client> = {};
```

Comment:
Here is an idea for further simplifying the common case for customers:

```
export const {
  getMany: getManyClients,
} = generatePresence<Client>("client");

let presentClients: Client[] = [];
r.subscribeWithPresence(
  async (tx, presentClientIDs: ClientID[]) => {
    // Could we overload generatePresence method to take a 
    // LookupID | ClientID to avoid having to do this map
    return await getManyClients(tx, presentClientIDs.map(clientID => ({clientID}));
  },
  (result) => {
    presentClients = result;
    render();
  },
);

function render() {
  for (const presentClient of presentClients) {
    renderCursor(presentClient);
  }
}
```

We would also want to provide a similar convenient merged helper for the React hooks.

Reply:
I think only providing this in "replicache-react" would be sufficient to start with

Reply:
>     // Could we overload generatePresence method to take a 
>     // LookupID | ClientID to avoid having to do this map
>     return await getManyClients(tx, presentClientIDs.map(clientID => ({clientID}));

At an early stage I had two functions. One for the case where there is no id and one for where the id is present.

The problem is that we are trying to solve 2 things with one function. Supporting multiple IDs per clientID is something we could get rid of and it would simplify things. It would mean that you cannot use rails for cases where each client has a collection of entities (multitouch pointers for example).

Maybe we should just do generatePresence where there is no ID and we can add genererateMultiPresence in the future if/when there is a demand.